wakuv2/waku.go

```diff
@@ -139,13 +140,21 @@ func New(nodeKey string, cfg *Config, logger *zap.Logger) (*Waku, error) {
 		return nil, fmt.Errorf("failed to setup the network interface: %v", err)
 	}
 
+	connStatusChan := make(chan node.ConnStatus)
+
 	waku.node, err = node.New(context.Background(),
 		node.WithPrivateKey(privateKey),
 		node.WithHostAddress([]net.Addr{hostAddr}),
 		node.WithWakuRelay(wakurelay.WithMaxMessageSize(int(waku.settings.MaxMsgSize))),
 		node.WithWakuStore(false), // Mounts the store protocol (without storing the messages)
+		node.WithConnStatusChan(connStatusChan),
 	)
 
+	go func() {
+		for c := range connStatusChan {
+			signal.SendPeerStats(c)
+		}
+	}()
```

Comment:
I think you'll have an import cycle issue, not sure signals should be emited in this file.
Also the go routine is never stopped. Perhaps something like this could be used? (there are similar examples of code in the repo)

```
go func(){
  for {
    select {
       case <-w.quit:
	   return
	case c <-connStatusChan:
           signal.SendPeerStats(c)
      }
}
}()
```

Reply:
Fixed the goroutine, thanks for pointing this out. What will the import cycle issue entail? Are you suggesting moving signals part to a separate module?

Reply:
The build will not complete due to failing in the lint step:
```
[2021-07-26T16:10:50.760Z] + make lint
[2021-07-26T16:10:50.760Z] lint
[2021-07-26T16:11:03.479Z] level=warning msg="[runner] Can't run linter goanalysis_metalinter: deadcode: analysis skipped: errors in package: [-: import cycle not allowed in test]"
[2021-07-26T16:11:06.430Z] level=warning msg="[runner] Can't run linter unused: buildir: failed to load package ens: could not load export data: no export data for \"github.com/status-im/status-go/protocol/ens [github.com/status-im/status-go/protocol/common.test]\""
[2021-07-26T16:11:06.430Z] level=error msg="Running error: buildir: failed to load package ens: could not load export data: no export data for \"github.com/status-im/status-go/protocol/ens [github.com/status-im/status-go/protocol/common.test]\""
[2021-07-26T16:11:06.430Z] make: *** [Makefile:304: lint] Error 3
script returned exit code 2
```

I had this issue with a different status-go PR https://github.com/status-im/status-go/pull/2286/commits  and had to reorganize the code just to get past that issue. You can see here in the highlighted code https://github.com/status-im/status-go/pull/2286/commits/ea3038a10b92a264529b7724539442715b5bc326#diff-bf66678b3e92c93f707a87660f5b33bf6b46d9a26c5eca024e816b1da20f3b62L199-L211 how I had to remove the go routine from waku.go and move it to a different layer. Probably something similar will be needed here

Reply:
Ah, ok, strange though, i see that Jenkins built the PR fine, also didn't see the problem appearing with `make statusgo` and with status-react builds.

Reply:
Cryptic error message from `make lint` in any case.